ci: enable sentry in the release builds

markoburcul opened the pull request:
Referenced issue: https://github.com/status-im/status-desktop/issues/16842

### What does the PR do

Adds Sentry related secrets to env vars for release builds.

### Affected areas

Desktop release build.

### Architecture compliance

- [ ] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

<!-- screenshot (or gif/video) that demonstrates the functionality, specially important if it's a bug fix. -->

<!-- Uncomment this section for status-go upgrade/dogfooding pull requests

### Impact on end user

None.

### How to test

Run the release pipeline and inspect from the Sentry UI if anything from the CI is being sent to it.

### Risk 

Leaking sentry secrets within pipeline.

Tick **one**:
- [ ] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

-->

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | cb9c9f0d | [#1](https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-16845/1/) | 2024-11-29 09:58:57 | ~7 min | `macos/aarch64` | [:apple:`dmg`](https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-241129-095115-cb9c9f-pr16845-aarch64.dmg) |

jakubgs commented on ci/Jenkinsfile.combined:
```diff
@@ -32,6 +32,13 @@ pipeline {
+      environment {
+        SENTRY_PRODUCTION = utils.isReleaseBuild() ? true : false
+      }
+      withCredentials([
+        string(credentialsId: 'status-desktop-sentry-dsn', variable: 'STATUS_BUILD_SENTRY_DSN_STATUS_DESKTOP'),
+        string(credentialsId: 'status-go-sentry-dsn', variable: 'STATUS_BUILD_SENTRY_DSN_STATUS_GO')
+      ])
```
Why are you adding this here? It's already done in `status-jenkins-lib`.

igor-sirotin reviewed (approved):
Nice, thank you!
I suppose `utils.isReleaseBuild()` is defined externally and distringuishes `ci.infra.status.im` from `ci.status.im`, right?

markoburcul commented:
> I suppose `utils.isReleaseBuild()` is defined externally and distringuishes `ci.infra.status.im` from `ci.status.im`, right?

Yes, it is defined in the jenkins library and returns true if one of the following conditions is met:
```
    params.RELEASE == true ||
    params.BUILD_TYPE == 'release' ||
    env.JOB_NAME.startsWith('status-mobile/release') ||
    env.JOB_NAME.startsWith('status-desktop/release') ||
    env.JOB_NAME.startsWith('status-node/release')
```

igor-sirotin commented:
Tested MacOS build from here: https://ci.infra.status.im/job/status-desktop/job/release/job/release%2Fsentry-test-panic-button/1/
Which has the `Trigger panic` button from here: https://github.com/status-im/status-desktop/pull/16850

Sentry report received ✅ 
<img width="687" alt="image" src="https://github.com/user-attachments/assets/220e9be6-3041-45a5-b33e-e84f334b445a">

But the environment is not set to `production` 🤔 
Will double check where's the error with this.

igor-sirotin commented:
@markoburcul I've fixed the issue and updated the branch.
Can you please start another release build from https://github.com/status-im/status-desktop/pull/16850? 

fix here: https://github.com/status-im/status-go/pull/6149

jakubgs commented on ci/Jenkinsfile.combined:
```diff
@@ -32,6 +32,9 @@ pipeline {
   stages {
     stage('Build') {
+      environment {
+        SENTRY_PRODUCTION = "${utils.isReleaseBuild() ? 'true' : 'false'}"
```
Why is this in the combined `Jenkinsfile`? It will not affect sub-jobs spawned by this job.

markoburcul commented:
> @markoburcul I've fixed the issue and updated the branch. Can you please start another release build from #16850?
> 
> fix here: [status-im/status-go#6149](https://github.com/status-im/status-go/pull/6149)

The build with newest updates: https://ci.infra.status.im/job/status-desktop/job/release/job/release%252Fsentry-test-panic-button/2/

igor-sirotin commented:
> The build with newest updates: https://ci.infra.status.im/job/status-desktop/job/release/job/release%252Fsentry-test-panic-button/2/

Thank you @markoburcul! I've tested the new build.

## TLDR

PR is good to go, thank you 🎉 
But would be nice to set up Github Integration in Sentry.

## Fixed things

1. `production` environment is now there ✅ 

    <img width="414" alt="image" src="https://github.com/user-attachments/assets/811e7c09-e26d-4469-af6c-1c883edd7258">

2. Stacktrace is properly there ✅ 
    
    <img width="1021" alt="image" src="https://github.com/user-attachments/assets/93b2ec4a-2a7b-4082-a997-32b268b5f270">
    
## Minor issue

I noted that the stacktrace doesn't contain code snippets at all 🤔 

For local crash reports, stacktrace looks like this:

<img width="673" alt="image" src="https://github.com/user-attachments/assets/de44d422-9d9e-43ac-86a0-97b61351b073">

Because each stacktrace frame contains a code snippet in the report:
    
<img width="871" alt="image" src="https://github.com/user-attachments/assets/7ecbc05d-36d7-4fcb-90c9-9b583f516618">

But it's not the case for release builds.

I believe it's **better**, because it uses less network data for users.
Though it's handy to have, I wonder if Sentry Github integration will fix this on our side.

@markoburcul can you please enable it? At least for the `status-go` repository.

<img width="932" alt="image" src="https://github.com/user-attachments/assets/7001485e-e97e-4204-bf4a-1b36a720fabb">

markoburcul commented:
> @markoburcul can you please enable it? At least for the `status-go` repository.

I get 404 page not found on GitHub when I try to add integration. Maybe its because I lack rights on our organization/enterprise. Maybe @jakubgs can try it?

jakubgs commented:
It won't work just by itself. For an integration with GitHub to work our Sentry instance needs to be registered as a GitHub App:
https://develop.sentry.dev/integrations/github/

jakubgs commented on ci/Jenkinsfile.linux:
```diff
@@ -73,6 +73,7 @@ pipeline {
     STATUS_CLIENT_TARBALL = "pkg/${utils.pkgFilename(ext: 'tar.gz', arch: getArch())}"
     /* prevent sharing cache dir across different jobs */
     GO_GENERATE_FAST_DIR = "${env.WORKSPACE_TMP}/go-generate-fast"
+    SENTRY_PRODUCTION = "${utils.isReleaseBuild() ? 'true' : 'false'}"
```
How is this env variable actually checked? Are we sure a `false` string won't just be treated as true?

igor-sirotin replied:
It's checked for `true` and `1`:
https://github.com/status-im/status-go/blob/987a9e8707e97c1757f87f621c036d671bb046fe/internal/sentry/params.go#L34

markoburcul commented:
> It won't work just by itself. For an integration with GitHub to work our Sentry instance needs to be registered as a GitHub App:
> https://develop.sentry.dev/integrations/github/

I don't have enough permissions on Status org to add the Github App for Sentry..

igor-sirotin commented:
@markoburcul @jakubgs I opened a separate issue to track this: 
- https://github.com/status-im/status-go/issues/6152
